cassandra/src/cassandra/collections/map.clj

```diff
@@ -1,101 +1,82 @@
+        (alia/execute conn (insert :maps (values [[:id 0]
+                                                  [:elements {}]]))))))
 
   (invoke! [this test op]
-    (cql/use-keyspace conn "jepsen_keyspace")
-    (case (:f op)
-      :add (try (cassandra/execute
-                  conn
-                  (str "UPDATE maps SET elements = elements + {"
-                       (:value op) " : " (:value op)
-                       "} WHERE id = 0;")
-                  :consistency-level (consistency-level writec))
-                (assoc op :type :ok)
-                (catch UnavailableException e
-                  (assoc op :type :fail :value (.getMessage e)))
-                (catch WriteTimeoutException e
-                  (assoc op :type :info :value :timed-out))
-                (catch NoHostAvailableException e
-                  (info "All nodes are down - sleeping 2s")
-                  (Thread/sleep 2000)
-                  (assoc op :type :fail :value (.getMessage e))))
-      :read (try (wait-for-recovery 30 conn)
-                 (let [value (->> (cassandra/execute
-                                    conn
-                                    "SELECT * from maps WHERE id = 0;"
-                                    :consistency-level (consistency-level :all)
-                                    :retry-policy aggressive-read)
-                                  first
-                                  :elements
-                                  vals
-                                  (into (sorted-set)))]
-                   (assoc op :type :ok :value value))
-                 (catch UnavailableException e
-                   (info "Not enough replicas - failing")
-                   (assoc op :type :fail :value (.getMessage e)))
-                 (catch ReadTimeoutException e
-                   (assoc op :type :fail :value :timed-out)))))
+    (try
+      (alia/execute conn (use-keyspace :jepsen_keyspace))
+      (case (:f op)
+        :add (do (alia/execute conn
+                               (update :maps
+                                       (set-columns {:elements [+ {(:value op) (:value op)}]})
+                                       (where [[= :id 0]]))
+                               {:consistency-level writec})
+                 (assoc op :type :ok))
+        :read (do (wait-for-recovery 30 conn)
+                  (let [value (->> (alia/execute conn
+                                                 (select :maps (where [[= :id 0]]))
+                                                 {:consistency  :all
+                                                  :retry-policy aggressive-read})
+                                   first
+                                   :elements
+                                   vals
+                                   (into (sorted-set)))]
+                    (assoc op :type :ok, :value value))))
+
+      (catch ExceptionInfo e
+        (let [e (class (:exception (ex-data e)))]
+          (cond
```

Comment:
Nice! Thank you. That cleans it up. I was always a bit confused on how to use `condp` because I think all the examples I had seen used `<` or `>`, which made it a bit confusing to parse where the argument was going.